includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -899,6 +896,15 @@ private function process_payment_with_payment_method( int $order_id ) {
 			$upe_payment_method            = $this->payment_methods[ $selected_payment_type ] ?? null;
 			$response_args                 = [];
 
+			if ( $this->spe_enabled && isset( $payment_method_details->type ) ) {
+				foreach ( self::UPE_AVAILABLE_METHODS as $payment_method_class ) {
+					$payment_method = new $payment_method_class();
+					if ( $payment_method->get_id() === $payment_method_details->type ) {
+						$upe_payment_method = $payment_method;
+					}
+				}
```

Comment:
```suggestion
    const UPE_AVAILABLE_METHOD_CLASSES = [
        'card' => WC_Stripe_UPE_Payment_Method_CC::class,
        'ideal' => '...'
     ];
	$upe_payment_method = new self::UPE_AVAILABLE_METHOD_CLASSES[ $payment_method_details->type ]
```

Instead of instantiating all the payment methods, how about we store the classes in an object and instantiate that class only? Something similar to this

Reply:
Yeah, this is much smarter, thanks! I did it here: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4152 to avoid changing the next PR since you reviewed it already